augur/ancestral.py

```diff
@@ -70,6 +70,16 @@ def run(args):
         print("ERROR: reading tree from %s failed."%args.tree)
         return 1
 
+    import numpy as np
+    missing_internal_node_names = [n.name is None for n in T.get_nonterminals()]
+    if np.all(missing_internal_node_names):
+        print("\n*** WARNING: Tree has no internal node names!")
+        print("*** Without internal node names, ancestral sequences can't be linked up to the correct node later.")
+        print("*** If you want to use 'augur export' or `augur translate` later, re-run this command with the output of 'augur refine'.")
+        print("*** If you haven't run 'augur refine', you can add node names to your tree by running:")
+        print("*** augur refine --tree %s --output-tree <filename>.nwk"%(args.tree) )
+        print("*** And use <filename>.nwk as the tree when running 'ancestral', 'translate', and 'traits'")
```

Comment:
I think this might raise a bigger question of how we want these individual parts to operate - more independently or always together? Should it be ok just to run `ancestral` or `translate` to get some information, without the intention of putting it together or through `export` (in which case node names don't matter), or do we want to make people always keep the pieces so that they *could* be put together, because they make that decision later without remembering the warnings? I can think for both sides, myself - it would be good to hear thoughts of others. (Should I ask in Slack?)

Reply:
I'm all for being able to use these features as independent parts, with no intention to run `export`. But if one runs `augur traits` (e.g.) using a tree without internal node names, then you get an output (json) which refers to things like _NODE00001_ which do not exist in the tree. I think in these cases we want to exit and say "We can only infer things onto a tree with internal node names labelled, else you cannot match the results to your tree! One way to do this is to run `augur refine --tree A --output-tree B`"

Reply:
I'm happy to change to this. Feel free to nudge me if I haven't done it in a few days...!